[D1] Get started workflow through Dashboard (#16856)
cloudflare/cloudflare-docs: src/content/docs/d1/get-started.mdx

:::

</TabItem> <TabItem label='Dashboard'>

<Steps>
1. Log in to your [Cloudflare dashboard](https://dash.cloudflare.com/sign-up/workers-and-pages).

> 2. From the menu, go to your account > **Workers & Pages** > **Overview**.

Reviewer: ```suggestion
2. Go to your account > **Workers & Pages** > **Overview**.
```

3. Select **Create**.
4. Select **Create Worker**.

> 5. Name for your Worker. For this tutorial, name your Worker "d1-tutorial".

Reviewer: ```suggestion
5. Name for your Worker. For this tutorial, name your Worker `d1-tutorial`.
```

:::

</TabItem> <TabItem label='Dashboard'>

<Steps>
1. Go to **Storage & Databases** > **D1**.

> 2. Select **+ Create**.

Reviewer: ```suggestion
2. Select **Create**.
```

> 3. Name your database. For this tutorial, name your D1 database "prod-d1-tutorial".

Reviewer: ```suggestion
3. Name your database. For this tutorial, name your D1 database `prod-d1-tutorial`.
```

> 4. (Optional) Provide a location hint. Location hint is an optional parameter you can provide to indicate your desired geographical location for your database. See [Provide a location hint](/d1/configuration/data-location/#provide-a-location-hint).

Reviewer: ```suggestion
4. (Optional) Provide a location hint. Location hint is an optional parameter you can provide to indicate your desired geographical location for your database. Refer to [Provide a location hint](/d1/configuration/data-location/#provide-a-location-hint) for more information.
```

Reviewer: We should avoid verbs such as "see" for inclusivity reasons.

You can also bind your D1 database to a [Pages Function](/pages/functions/). For more information, refer to [Functions Bindings for D1](/pages/functions/bindings/#d1-databases).

</TabItem> <TabItem label='Dashboard'>

You create bindings by adding them to the Worker you have created.

<Steps>

> 1. From the menu, go to your account > **Workers & Pages** > **Overview**.

Reviewer: ```suggestion
1. Go to **Workers & Pages** > **Overview**.
```

Reviewer: Since the above steps also assume the customer is already in the CF dash and their account, I'm removing this for consistency. Not sure about the workflow and if we need to add the "log in to cf dash and choose account" in all 1st steps on this page, though

> 2. Select the "d1-tutorial" Worker you created in [step 1](/d1/get-started/#1-create-a-worker).

Reviewer: ```suggestion
2. Select the `d1-tutorial` Worker you created in [step 1](/d1/get-started/#1-create-a-worker).
```

> 3. Go to the **Settings** tab.

Reviewer: ```suggestion
3. Select **Settings**.
```

> 4. Scroll to **Bindings**, then select **+ Add**.

Reviewer: ```suggestion
4. Scroll to **Bindings**, then select **Add**.
```

> 5. From the **Add a resource binding** panel, select **D1 database**.

Reviewer: ```suggestion
5. Select **D1 database**.
```

Reviewer: It's more direct. Also, when you say "from the..." it makes it seem there's an option on that panel from which we then need to choose D1

> 6. Name your binding in the **Variable name** field, then select the "prod-d1-tutorial" D1 database you created in [step 2](/d1/get-started/#2-create-a-database) from the dropdown menu. For this tutorial, name your binding "DB".

Reviewer: ```suggestion
6. Name your binding in **Variable name**, then select the `prod-d1-tutorial` D1 database you created in [step 2](/d1/get-started/#2-create-a-database) from the dropdown menu. For this tutorial, name your binding `DB`.
```

</Steps>

</TabItem> <TabItem label='Dashboard'>

Use the Dashboard to create a table and populate it with data.

<Steps>
1. Go to **Storage & Databases** > **D1**.

> 2. Select the "prod-d1-tutorial" database you created in [step 2](/d1/get-started/#2-create-a-database).

Reviewer: ```suggestion
2. Select the `prod-d1-tutorial` database you created in [step 2](/d1/get-started/#2-create-a-database).
```

3. Select **Create table**.
4. Name your table. For this tutorial, create a table called "Customers".

> 5. Create three columns (add more columns by selecting **+ Add Column**):

Reviewer: ```suggestion
5. Create three columns (add more columns by selecting **Add Column**):
```

	1. Column Name: `CustomerId`, Type: `integer`, Default Value: `NULL` (leave blank). Set this column as a primary key.
	2. Column Name: `CompanyName`, Type: `text`, Default Value: `NULL` (leave blank).
	3. Column Name: `ContactName`, Type: `text`, Default Value: `NULL` (leave blank).
6. Select **Create**.
7. Select the **Customers** table.

> 8. Select **Add data** to populate the columns of the table. Select **Save** to add the row to the table. Add the following rows:

Reviewer 2: Could we maybe provide also another option here to just copy paste the same SQL commands as in the CLI version using the `Console` tab in the UI dashboard? Less steps, and without many differences between the two guides.

Author: Very good suggestion! Will implement, thanks @lambrospetrou

After configuring your Worker, you can test your project locally before you deploy globally.

</TabItem> <TabItem label='Dashboard'>

You can query your D1 database using your Worker.

<Steps>

> 1. Go to your account > **Workers & Pages** > **Overview**.

Reviewer: ```suggestion
1. Go to **Workers & Pages** > **Overview**.
```

> 2. Select the "d1-tutorial" Worker you created.

Reviewer: ```suggestion
2. Select the `d1-tutorial` Worker you created.
```

3. Select **Edit Code**.
4. Clear the contents of the `worker.js` file, then paste the following code:

	```js
	export default {
		async fetch(request, env) {
			const { pathname } = new URL(request.url);

			if (pathname === "/api/beverages") {
				// If you did not use `DB` as your binding name, change it here
				const { results } = await env.DB.prepare(
					"SELECT * FROM Customers WHERE CompanyName = ?"
				)
					.bind("Bs Beverages")
					.all();
				return new Response(JSON.stringify(results), {
					headers: { 'Content-Type': 'application/json' }
				});
			}

			return new Response(
				"Call /api/beverages to see everyone who works at Bs Beverages"
			);
		},
	};
	```
5. Select **Save**.

</Steps>
</TabItem>
</Tabs>

## 5. Deploy your database

Deploy your database on Cloudflare's global network.

<Tabs syncKey='CLIvDash'> <TabItem label='CLI'>

> To deploy your Worker to production using Wrangler, you must first repeat the [database bootstrapping](/d1/get-started/#configure-your-d1-database) steps after replacing the `--local` flag with the `--remote` flag to give your Worker data to read. This creates the database tables and imports the data into the production version of your database.

Reviewer 2: ```suggestion
To deploy your Worker to production using Wrangler, you must first repeat the [database configuration](/d1/get-started/#configure-your-d1-database) steps after replacing the `--local` flag with the `--remote` flag to give your Worker data to read. This creates the database tables and imports the data into the production version of your database.
```

	```sh output
	Outputs: https://d1-tutorial.<YOUR_SUBDOMAIN>.workers.dev
	```

	You can now visit the URL for your newly created project to query your live database.

	For example, if the URL of your new Worker is `d1-tutorial.<YOUR_SUBDOMAIN>.workers.dev`, accessing `https://d1-tutorial.<YOUR_SUBDOMAIN>.workers.dev/api/beverages` sends a request to your Worker that queries your live database directly.

4. Test your database is running successfully. Add `/api/beverages` to the provided Wrangler URL. For example, `https://d1-tutorial.<YOUR_SUBDOMAIN>.workers.dev/api/beverages`.

</Steps>

</TabItem> <TabItem label='Dashboard'>
<Steps>

> 1. Go to your account > **Workers & Pages** > **Overview**.

Reviewer: ```suggestion
1. Go to **Workers & Pages** > **Overview**.
```

> 2. Select your "d1-tutorial" Worker.

Reviewer: ```suggestion
2. Select your `d1-tutorial` Worker.
```

> 3. Go to the **Deployments** tab.

Reviewer: ```suggestion
3. Select **Deployments**.
```

4. From the **Version History** table, select **Deploy version**.
5. From the **Deploy version** page, select **Deploy**.

</Steps>

This deploys the latest version of the Worker code to production.

</TabItem></Tabs>

## 6. (Optional) Develop locally with Wrangler

> If you are using D1 with Wrangler, you can test your database locally. While in your project directory:

Reviewer 2: This might be nuance, but here we are testing the "worker", not the database. The local database was already tested with the `--local` flag. I suggest being explicit.

<Steps>
1. Run `wrangler dev`:

	```sh
	npx wrangler dev
	```

	When you run `wrangler dev`, Wrangler provides a URL (most likely `localhost:8787`) to review your Worker.

> 2. Navigate to the URL.

Reviewer: ```suggestion
2. Go to the URL.
```

	The page displays `Call /api/beverages to see everyone who works at Bs Beverages`.

3. Test your database is running successfully. Add `/api/beverages` to the provided Wrangler URL. For example, `localhost:8787/api/beverages`.
</Steps>

If successful, the browser displays your data.

:::note
You can only develop locally if you are using Wrangler. You cannot develop locally through the Cloudflare dashboard.
:::

## 7. (Optional) Delete your database

> To delete your **database**:

Reviewer: Any particular reason why this is bold? We only use bold for specific actions

<Tabs syncKey='CLIvDash'> <TabItem label='CLI'>

Run:

```sh
npx wrangler d1 delete prod-d1-tutorial
```

</TabItem><TabItem label='Dashboard'>

<Steps>

> 1. Go to your account > **Storages & Databases** > **D1**.

Reviewer: ```suggestion
1. Go to **Storages & Databases** > **D1**.
```

> 2. Select your "prod-d1-tutorial" D1 database.

Reviewer: ```suggestion
2. Select your `prod-d1-tutorial` D1 database.
```

> 3. Go to the **Settings** tab.

Reviewer: ```suggestion
3. Select **Settings**.
```

4. Select **Delete**.

> 5. Type the name of the database ("prod-d1-tutorial") to confirm the deletion.

Reviewer: ```suggestion
5. Type the name of the database (`prod-d1-tutorial`) to confirm the deletion.
```

</Steps>

</TabItem> </Tabs>

If you want to delete your **Worker**:

<Tabs syncKey='CLIvDash'> <TabItem label='CLI'>

Run:

```sh
npx wrangler delete d1-tutorial
```

</TabItem> <TabItem label='Dashboard'>

<Steps>

> 1. Go to your account > **Workers & Pages** > **Overview**.

Reviewer: ```suggestion
1. Go to **Workers & Pages** > **Overview**.
```

> 2. Select your "d1-tutorial" Worker.

Reviewer: ```suggestion
2. Select your `d1-tutorial` Worker.
```

> 3. Go to the **Settings** tab.

Reviewer: ```suggestion
3. Select **Settings**.
```

4. Scroll to the bottom of the page, then select **Delete**.

> 5. Type the name of the Worker ("d1-tutorial") to confirm the deletion.

Reviewer: ```suggestion
5. Type the name of the Worker (`d1-tutorial`) to confirm the deletion.
```
